grpc-api-client/src/main/java/io/deephaven/grpc_api_client/table/BarrageSourcedTable.java

```diff
@@ -430,18 +420,22 @@ private void processPendingFrees() {
     }
 
     private void doFreeRows(final Index rowsToFree) {
-        final TLongArrayList freedRows = new TLongArrayList(rowsToFree.intSize());
+        // Note: these are NOT OrderedKeyIndices until after the call to .sort()
+        final WritableLongChunk<Attributes.OrderedKeyIndices> redirectedRows
+                = WritableLongChunk.makeWritableChunk(rowsToFree.intSize("BarrageSourcedTable"));
+        redirectedRows.setSize(0);
 
         for (final Index.Iterator removedIt = rowsToFree.iterator(); removedIt.hasNext();) {
             final long next = removedIt.nextLong();
             final long prevIndex = redirectionIndex.remove(next);
             if (prevIndex == -1) {
                 Assert.assertion(false, "prevIndex != -1", prevIndex, "prevIndex", next, "next");
             }
-            freedRows.add(prevIndex);
+            redirectedRows.add(prevIndex);
         }
 
-        freeset.insert(Index.FACTORY.getIndexByValues(freedRows));
+        redirectedRows.sort(); // now they're truly ordered
+        freeset.insert(redirectedRows, 0, redirectedRows.size());
     }
 
     @Override
```

Comment:
There are pending frees anymore ever.

Reply:
And maybe we don't need `freeze` anymore? Either way, as used `frozen` need not be `volatile`.